### PR TITLE
feat(wallet): ENS Off Chain Lookup in New Send Tab

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Implemented `ENS Off Chain Lookup` for the new `Send` tab

Resolves <https://github.com/brave/brave-browser/issues/27109>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### Testing Ethereum
1. Go to brave://settings/web3 and make sure `Allow ENS offchain lookup` is set to `Ask`
2. Go to brave://wallet/send, choose a token on the `Ethereum` network, enter an amount to send and paste in `offchainexample.eth`
3. You should see the `ENS Offchain` message
4. Make sure the `Learn more` link routes to https://github.com/brave/brave-browser/wiki/ENS-offchain-lookup
5. Click on the `Use ENS domain` button
6. The `Message` should disappear and resolve the address.

https://user-images.githubusercontent.com/40611140/207937782-89e24c56-e7f7-4580-83a0-351e27e73e7a.mov

### CODE REVIEW

**Review by josheleonard — CHANGES_REQUESTED** [review]

**Review by darkdh — APPROVED** [review]

File: components/brave_wallet_ui/common/hooks/send.ts [hunk 1]
```diff
@@ -473,6 +478,7 @@ export default function useSend (isSendTab?: boolean) {
     sendAmountValidationError,
     showEnsOffchainLookupOptions,
     ensOffchainLookupOptions,
+    setShowEnsOffchainLookupOptions,
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/common/hooks/send.ts:482]: Is now wrapped in a `useCallback` hook!

File: components/brave_wallet_ui/page/screens/send/components/address-message/address-message.style.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at https://mozilla.org/MPL/2.0/.
+
+import styled from 'styled-components'
+
+export const LearnMoreButton = styled.a`
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/page/screens/send/components/address-message/address-message.style.ts]: Updated!

File: components/brave_wallet_ui/page/screens/send/send/send.tsx [hunk 3]
```diff
@@ -207,6 +219,13 @@ export const Send = (props: Props) => {
     return undefined
   }, [selectedSendAsset, networks])
 
+  const hasAddressError = React.useMemo(() => {
+    return searchingForDomain
+      ? false
+      : (addressError !== undefined && addressError !== '') ||
+      (addressWarning !== undefined && addressWarning !== '')
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/page/screens/send/send/send.tsx]: Updated!

File: components/brave_wallet_ui/page/screens/send/send/send.tsx [hunk 4]
```diff
@@ -219,6 +238,19 @@ export const Send = (props: Props) => {
...
 
+  const addressMessageInformation: AddressMessageInfo | undefined = React.useMemo(() => {
+    // ToDo: Implement Invalid Checksum warning and other longer warnings here in the future.
+    // https://github.com/brave/brave-browser/issues/26957
+    if (showEnsOffchainLookupOptions) {
+      return {
+        title: getLocale('braveWalletEnsOffChainLookupTitle'),
+        description: getLocale('braveWalletEnsOffChainLookupDescription'),
+        url: 'https://github.com/brave/brave-browser/wiki/ENS-offchain-lookup'
+      }
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/page/screens/send/send/send.tsx]: Moved to `send-ui-messages.ts` file!

**Review by josheleonard — APPROVED** [review]
++

**Pull request merged**